CMake/build.cmake

```diff
@@ -13,23 +15,37 @@ if(NOT CMAKE_BUILD_TYPE)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+# SENSEI should always be position independent
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+# SENSEI requires minimum C++11
+if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 11)
+  set(CMAKE_CXX_STANDARD 11)
+endif ()
+
 if (NOT MSVC)
-  if (NOT CMAKE_CXX_FLAGS)
-  set(tmp "-fPIC -std=c++11 -Wall -Wextra")
+  string(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra")
+
+  # Need to explicitly set optimization level to -O3 because some systems default to -O2
+  set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
+  set(optimization_flags " -mtune=native; -march=native")
```

Comment:
Please make this an option so we can disable it in package management.

Seting march native does not work with:
- most non-gcc compiler, even on Unix
- Spack: the package manager will set the architecture via env control
- HPC systems with different login/compute nodes: there we hint the arch ourselves via env vars

Reply:
`-march=native -mtune=native` is supported for intel, gcc and clang. Do we need to support other compilers? Which other compilers do you want to see supported?

Reply:
which env vars do you use? Note: that the CXXFLAGS env var takes precedence over our defaults.